### PR TITLE
Various usNIC cleanups and fixes

### PR DESCRIPTION
**Pull request by jsquyres**
1. Remove dead / stale / unsupported code
1. Change behavior to _always_ return a domain name, even when the API version is less than v1.4 (see commit message / comments for further description)
1. Make sure to obey AV hints

### CODE REVIEW

File: prov/usnic/src/usdf_domain.c [hunk 1]
```diff
@@ -410,46 +336,23 @@ bool usdf_domain_checkname(uint32_t version, struct usd_device_attrs *dap,
...
-
-	if (version) {
-		valid = false;
-
-		ret = usdf_domain_getname(version, dap, &reference);
-		if (ret < 0)
-			return false;
+        /* If no hint was provided, then the check is happy/valid/succeeded */
+	if (!hint)
+		return true;
```
Comment (rfaucett) [inline prov/usnic/src/usdf_domain.c]: No braces?  Also, "NULL == hint" - hint is not a boolean, it is a pointer.

File: prov/usnic/src/usdf_domain.c [hunk 2]
```diff
@@ -410,46 +336,23 @@ bool usdf_domain_checkname(uint32_t version, struct usd_device_attrs *dap,
...
-	}
-
-	/* If hint is non-NULL then assume the version is 1.4 if not provided.
-	 */
-	if (hint)
-		return usdf_domain_checkname(FI_VERSION(1, 4), dap, hint);
+	valid = (strcmp(reference, hint) == 0);
+	if (!valid)
+	USDF_DBG("given hint %s does not match %s -- invalid\n",
+		hint, reference);
```
Comment (rfaucett) [inline prov/usnic/src/usdf_domain.c]: No braces AND no indent?

File: prov/usnic/src/usdf_ep_dgram.c [hunk 3]
```diff
@@ -508,6 +508,7 @@ int usdf_dgram_fill_dom_attr(uint32_t version, const struct fi_info *hints,
 
 	defaults = dgram_dflt_domain_attr;
 	ret = usdf_domain_getname(version, dap, &defaults.name);
+        printf("Got USDF DGRAM domain name: %s (ret=%d)\n", defaults.name, ret);
```
Comment (rfaucett) [inline prov/usnic/src/usdf_ep_dgram.c]: indent does not appear to match?

File: prov/usnic/src/usdf_domain.c [hunk 4]
```diff
@@ -365,43 +283,51 @@ usdf_domain_open(struct fid_fabric *fabric, struct fi_info *info,
...
+	char *buf;
 
-	if (FI_VERSION_GE(version, FI_VERSION(1, 4))) {
-		buf = strdup(dap->uda_devname);
-		if (!buf) {
-			ret = -errno;
-			USDF_DBG("strdup failed while creating domain name\n");
-		}
+	buf = strdup(dap->uda_devname);
+	if (!buf) {
```
Comment (rfaucett) [inline prov/usnic/src/usdf_domain.c]: NULL == buf

File: prov/usnic/src/usdf_domain.c [hunk 5]
```diff
@@ -365,43 +283,51 @@ usdf_domain_open(struct fid_fabric *fabric, struct fi_info *info,
...
+		ret = -errno;
+		USDF_DBG("strdup failed while creating domain name\n");
 	}
 
 	*name = buf;
```
Comment (rfaucett) [inline prov/usnic/src/usdf_domain.c]: should we really modify *name on failure?

File: prov/usnic/src/usdf_domain.c [hunk 6]
```diff
@@ -365,43 +283,51 @@ usdf_domain_open(struct fid_fabric *fabric, struct fi_info *info,
...
- *    Just do a string compare.
- * 2. Version is greater than or equal to 1.4 and provided hint is NULL.  Treat
- *    this as _valid_ as it could be an application requesting a 1.4 domain name
- *    but not providing an explicit hint.
- * 3. Version is less than 1.4 and a name hint is provided.  This should always
- *    be _invalid_.
- * 4. Version is less than 1.4 and name hint is NULL. This will always be
- *    _valid_.
+ * Hence, if a hint was provided, check the domain name (that we now
+ * always have) against the hint.
```
Comment (rfaucett) [inline prov/usnic/src/usdf_domain.c]: What does this routine do?  Why does no hint mean return true?

Comment (jsquyres) [reply prov/usnic/src/usdf_domain.c]: It checks to see if the domain name supplied in a hint matches our current domain name.  Added a comment line about this.

File: prov/usnic/src/usdf_domain.c (hunk 2 above)
Comment (jsquyres) [reply prov/usnic/src/usdf_domain.c]: Ooops; indenting fail.  Fixed.

FWIW: I stuck with the prior Linux-kernel-like style (no braces for 1-line blocks, `!foo` for checking for NULL, etc.).  But I have no strong preference here -- I updated all the places you noted per your comments.

File: prov/usnic/src/usdf_ep_dgram.c (hunk 3 above)
Comment (jsquyres) [reply prov/usnic/src/usdf_ep_dgram.c]: Oops -- spaces vs. tabs there.  Also, that's a debugging printf that slipped through.  Removed.

**Comment by jsquyres** [comment]
New commits pushed.

**Pull request merged**